pkg/disruptors/pod.go

```diff
@@ -68,10 +112,7 @@ func NewPodDisruptor(
 	}
 
 	// ensure selector and controller use default namespace if none specified
-	namespace := selector.Namespace
-	if selector.Namespace == "" {
-		selector.Namespace = metav1.NamespaceDefault
-	}
+	namespace := selector.NamespaceWithDefault()
```

Comment:
I find this name confusing. `With` methods are generally used in the disruptor codebase in builders. I gest something like `ForceNamespace` or something more specific of what this function is doing.

Reply:
Agreed, I'm not super sold on it. I'll try to think about something better.

Reply:
`NamespaceOrDefault` is the best thing I could come up with: https://github.com/grafana/xk6-disruptor/pull/267/commits/583e99227853cab5193d9e0ab954e0b53701b01e